usaspending_api/awards/v2/views/idvs/activity.py

```diff
@@ -105,20 +145,33 @@ def _business_logic(request_data: dict) -> tuple:
         # TinyShield.  We will either have an internal award id that is an
         # integer or a generated award id that is a string.
         award_id = request_data['award_id']
+        hide_edge_cases = request_data.get('hide_edge_cases')
+        hide_edges_awarded_amount = ''
+        hide_edges_end_date = ''
         award_id_column = 'award_id' if type(award_id) is int else 'generated_unique_award_id'
-
-        sql = COUNT_ACTIVITY_SQL.format(
-            award_id_column=Identifier(award_id_column),
-            award_id=Literal(award_id)
-        )
+        if hide_edge_cases:
+            hide_edges_awarded_amount = "and ca.base_and_all_options_value > 0"
+            hide_edges_end_date = "where tf.period_of_perf_potential_e is not null"
```

Comment:
Including the "where" in this segment is dangerous if future development involves adding a new clause that might end up above the date filter. I would advise putting the WHERE in the sql template.

Reply:
Wouldn't work if the clause were blank. This can be revisited if this query comes up again.

Reply:
if we put the `WHERE` in the sql clause, it will be blank when `hide_edge_cases` is false, which will cause an error.